cmd/apiserver-boot/boot/util/util.go

```diff
@@ -139,3 +142,16 @@ func CheckInstall() {
 			strings.Join(missing, ","))
 	}
 }
+
+func CancelWhenSignaled(parent context.Context) context.Context {
```

Comment:
reusing existing signal handler from here seems better to me : 

[https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/sample-controller/pkg/signals/signal.go#L29](https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/sample-controller/pkg/signals/signal.go#L29)

Reply:
@interma any plan to bump the thread?

Reply:
double-receiving should be a fine addition to this piece of code if that's what you mean, but the context is also used to support cancelling when one of the processes unexpectedly quits.

Reply:
> double-receiving should be a fine addition to this piece of code if that's what you mean

for now we're not catching signal in the code so there's no double-receiving. i mean the pull LGTM overall, but we should reuse the existing code from upstream as much as possible.

Reply:
I've attempted to address this final code-review comment in https://github.com/kubernetes-sigs/apiserver-builder-alpha/pull/410

It'd be great to get this fixed in the next release.

Reply:
Trouble is that importing  SetupSignalHandler from k8s.io/apiserver causes a clash with the k8s.io/kube-openapi dependency.
So I copied the code across instead.